File: rent-now-backend/src/main/java/pl/holowinska/rentnowbackend/model/rs/AccommodationRS.java
package pl.holowinska.rentnowbackend.model.rs;

import lombok.Builder;
import lombok.Data;
import pl.holowinska.rentnowbackend.model.enums.ConvenienceType;

import java.math.BigDecimal;
import java.util.HashMap;
import java.util.UUID;

@Data
@Builder
public class AccommodationRS {

    private Long id;
    private String name;
    private AddressRS addressRS;
    private UUID userUUID;
    private BigDecimal priceForDay;
    private BigDecimal squareFootage;
    private String description;
    private HashMap<ConvenienceType, BigDecimal> conveniences;
    private Integer maxNoOfPeople;
}

File: rent-now-ui/src/app/data/model/rs/AccommodationRS.ts
import {AddressRS} from "./AddressRS";

export interface AccommodationRS {
  id: number;
  addressRS: AddressRS;
  userUUID: string;
  name: string;
  priceForDay: number;
  squareFootage: number;
  description: string;
  conveniences: Map<string, number>;
  maxNoOfPeople: number;
}
